npeps/npep-173-service-account-matches.md

```diff
@@ -0,0 +1,99 @@
+# NPEP-173: NPEP template
+
+* Issue: [#173](https://github.com/kubernetes-sigs/network-policy-api/issues/173)
+* Status: Provisional
+
+## TLDR
+
+Ability to create policies that control network traffic based on workload identities a.ka.
+[Kubernetes service accounts](https://kubernetes.io/docs/concepts/security/service-accounts/).
+
+## Goals
+
+* Use service accounts (identities) as a way of describing (admin) network policies for pods
+
+## Non-Goals
+
+* There might be other identity management constructs like SPIFFE IDs which
+are outside the scope of this enhancement. We can only provide a way to select
+constructs known to core Kubernetes.
+
+## Introduction
+
+Every pod in Kubernetes will have an associated service account. If user does
+not explicitly provide a service account, a default service account is created
+in the namespace. A given pod will always have 1 service account associated
+with itself. So instead of using pod labels as a way of selecting pods sometimes
+there might be use cases to use service accounts as a way of selecting the pods.
+This NPEP tries to capture the design details for service accounts as selectors
+for policies.
+
+## User-Stories/Use-Cases
+
+1. **As a** cluster admin **I want** to select pods using their service accounts
+   instead of labels **so that** I can avoid having to setup webhooks and
+   validation admission policies to prevent users from changing labels on their
+   namespaces and pods that will make my policy start/stop matching the traffic
+   which is undesired
+2. **As a** cluster admin **I want** to select pods using their service accounts
+   instead of labels **so that** I can avoid the scale impact caused due to
+   mutation of pod/namespace labels that will cause a churn which makes my CNI
+   implementation react to that every time user changes a label.
+3. **As a** cluster admin my workloads have immutable identities and **I want**
+   to apply policies per workloads using their service accounts instead of labels
+   **since** I want to have eventual consistency of that policy in the cluster.
+
```

Comment:
hmm so this would be the non-goal I had but we can revisit this and I can try to add this user story, let me learn more about the istio use case here and see if for each of my above use cases I can come up with concrete examples?